File: 2026-02-February-GFG-POTD/February 22 - Count Subarrays with given XOR/Solution.java
class Solution {
    public long subarrayXor(int arr[], int k) {
        HashMap<Integer, Integer> map = new HashMap<>();
        
        long count = 0;
        int xor = 0;
        
        // Base case: prefix XOR 0 appears once
        map.put(0, 1);
        
        for (int num : arr) {
            xor ^= num;
            
            // If (xor ^ k) exists, add its frequency
            if (map.containsKey(xor ^ k)) {
                count += map.get(xor ^ k);
            }
            
            // Update frequency of current prefix xor
            map.put(xor, map.getOrDefault(xor, 0) + 1);
        }
        
        return count;
    }
}

File: 2026-02-February-GFG-POTD/February 22 - Count Subarrays with given XOR/Solution.py
class Solution:
    def subarrayXor(self, arr, k):
        from collections import defaultdict
        
        mp = defaultdict(int)
        
        count = 0
        xr = 0
        
        # Base case: prefix XOR 0 appears once
        mp[0] = 1
        
        for num in arr:
            xr ^= num
            
            # If (xr ^ k) exists, add its frequency
            count += mp[xr ^ k]
            
            # Update frequency of current prefix xor
            mp[xr] += 1
        
        return count
